localplanning.services/src/components/applications/ApplicationViewer.tsx

```diff
@@ -0,0 +1,43 @@
+import { useHtmlDownload } from "./hooks/useHTMLDownload";
+
+export const ApplicationViewer = () => {
+  const applicationId = new URLSearchParams(window.location.search).get("applicationId")
+  const { htmlContent, loading, error, fetchHtml } = useHtmlDownload(applicationId);
+
+  if (loading) {
+    return (
+      <div className="application-viewer">
+        <div className="loading-state">
+          <div className="spinner"></div>
+          <p>Loading application...</p>
+        </div>
+      </div>
+    );
+  }
+
+  if (error) {
+    return (
+      <div className="application-viewer">
+        <div className="error-state">
+          <h2>Failed to load application</h2>
+          <p>Error: {error}</p>
+          <button onClick={fetchHtml} className="retry-btn">
+            Retry
+          </button>
+        </div>
+      </div>
+    );
+  }
```

Comment:
This is pretty basic just now @ianjon3s - we don't really have a pattern here yet for loading and error states. Very open to ideas!